tests/integration_tests/tests/agentless_tests/test_executions.py

```diff
@@ -603,6 +604,31 @@ def test_cancel_execution_and_then_force_cancel(self):
 
         self._assert_execution_cancelled(execution, deployment_id)
 
+    def test_execute_and_kill_execution(self):
+        """Tests the kill execution option.
+
+        The function executes a new execution, writes the process to a local
+        file, kills the execution and asserts the process was also killed.
+        """
+        execution, node_inst_id, deployment_id = \
+            self._execute_from_resource(workflow_id='read_write',
+                                        workflow_params=None,
+                                        resource_file='dsl'
+                                                      '/read_write_workflows'
+                                                      '.yaml')
+        self._check_state('active_and_wrote', "Execution was expected to go "
```

Comment:
What if by the time we arrive to this line the status changes to `asleep`? 
Not sure if this is needed here..

Reply:
I think it is needed to make sure the pid was written as a runtime property, but I'll be glad to talk about it and see if there's maybe something I don't understand